File: app_ios/NoteFlow/NoteFlow/Models/Task.swift
//
//  Task.swift
//  NoteFlow
//
//  Created by toan on 11/6/25.
//

import Foundation
import SwiftData

@Model
final class Task {
    var id: UUID
    var title: String
    var content: String
    var priority: Priority
    var status: TaskStatus
    var tags: [String]
    var dueDate: Date?
    var createdAt: Date
    var updatedAt: Date
    
    init(title: String, content: String, priority: Priority = .medium, status: TaskStatus = .todo, tags: [String] = [], dueDate: Date? = nil) {
        self.id = UUID()
        self.title = title
        self.content = content
        self.priority = priority
        self.status = status
        self.tags = tags
        self.dueDate = dueDate
        self.createdAt = Date()
        self.updatedAt = Date()
    }
}

File: app_android/app/src/main/java/com/zplus/noteflow/data/model/Task.kt
package com.zplus.noteflow.data.model

import java.util.*

data class Task(
    val id: String,
    val title: String,
    val content: String,
    val priority: TaskPriority,
    val status: TaskStatus,
    val tags: List<String> = emptyList(),
    val dueDate: Date? = null,
    val assignee: String? = null,
    val createdAt: Date,
    val updatedAt: Date
)

enum class TaskPriority(val value: String) {
    LOW("low"),
    MEDIUM("medium"),
    HIGH("high"),
    CRITICAL("critical");
    
    companion object {
        fun fromString(value: String): TaskPriority {
            return values().find { it.value == value } ?: MEDIUM
        }
    }
}

enum class TaskStatus(val value: String) {
    TODO("todo"),
    IN_PROGRESS("in_progress"),
    REVIEW("review"),
    DONE("done");
    
    companion object {
        fun fromString(value: String): TaskStatus {
            return values().find { it.value == value } ?: TODO
        }
    }
}